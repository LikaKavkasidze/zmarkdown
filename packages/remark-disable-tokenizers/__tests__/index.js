import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import plugin from '../src/'

test('block throws + inline', () => {
  function t () {
    unified()
      .use(reParse)
      .use(plugin, {
        block: [
          ['blockquote', 'Blockquote are not allowed!'],
        ],
        inline: [
          'emphasis',
        ],
      })
      .use(remark2rehype)
      .use(stringify)
      .processSync(dedent`
        *a*

        > *a*

        foo \`bar\` baz
      `)
  }

  expect(t).toThrowErrorMatchingSnapshot()
})

test('block + inline', () => {
  const {contents} = unified()
    .use(reParse)
    .use(plugin, {
      block: [
        'blockquote',
      ],
      inline: [
        'emphasis',
      ],
    })
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(contents).toMatchSnapshot()
})

test('block', () => {
  const {contents} = unified()
    .use(reParse)
    .use(plugin, {
      block: [
        'blockquote',
      ],
    })
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(contents).toMatchSnapshot()
})

test('inline', () => {
  const {contents} = unified()
    .use(reParse)
    .use(plugin, {
      inline: [
        'emphasis',
      ],
    })
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(contents).toMatchSnapshot()
})

test('inline throws', () => {
  function t () {
    unified()
      .use(reParse)
      .use(plugin, {
        inline: [
          ['emphasis', 'nope'],
        ],
      })
      .use(remark2rehype)
      .use(stringify)
      .processSync(dedent`
        *a*

        > *a*

        foo \`bar\` baz
      `)
  }

  expect(t).toThrowErrorMatchingSnapshot()
})

test('block throws', () => {
  function t () {
    unified()
      .use(reParse)
      .use(plugin, {
        block: [
          ['blockquote', 'Blockquote are not allowed!'],
        ],
      })
      .use(remark2rehype)
      .use(stringify)
      .processSync(dedent`
        *a*

        > *a*

        foo \`bar\` baz
      `)
  }

  expect(t).toThrowErrorMatchingSnapshot()
})

test('block throws + inline throws', () => {
  function t () {
    unified()
      .use(reParse)
      .use(plugin, {
        block: [
          ['blockquote', 'Blockquote are not allowed!'],
        ],
        inline: [
          ['emphasis', 'Nope.'],
        ],
      })
      .use(remark2rehype)
      .use(stringify)
      .processSync(dedent`
        *a*

        > *a*

        foo \`bar\` baz
      `)
  }

  expect(t).toThrowErrorMatchingSnapshot()
})

test('does nothing', () => {
  const {contents} = unified()
    .use(reParse)
    .use(plugin)
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(contents).toMatchSnapshot()
})


test('unknown tokenizer', () => {
  const {contents} = unified()
    .use(reParse)
    .use(plugin, {
      inline: [
        'foo bar',
      ],
    })
    .use(remark2rehype)
    .use(stringify)
    .processSync(dedent`
      *a*

      > *a*

      foo \`bar\` baz
    `)

  expect(contents).toMatchSnapshot()
})

test('regression: actually turn off the tokenizer #412', () => {
  const {contents} = unified()
    .use(reParse)
    .use(plugin, {
      block: [
        ['html'],
      ],
      inline: [
        ['html'],
      ],
    })
    .use(remark2rehype)
    .use(stringify)
    .processSync(`hello\nworld`)

  expect(contents).toMatchSnapshot()
})
