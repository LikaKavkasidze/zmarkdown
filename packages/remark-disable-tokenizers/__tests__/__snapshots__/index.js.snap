// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block + inline 1`] = `
"<p>*a*</p>
<p>> *a*</p>
<p>foo <code>bar</code> baz</p>"
`;

exports[`block 1`] = `
"<p><em>a</em></p>
<p>> <em>a</em></p>
<p>foo <code>bar</code> baz</p>"
`;

exports[`block throws + inline 1`] = `"Blockquote are not allowed!"`;

exports[`block throws + inline throws 1`] = `"Blockquote are not allowed!"`;

exports[`block throws 1`] = `"Blockquote are not allowed!"`;

exports[`does nothing 1`] = `
"<p><em>a</em></p>
<blockquote>
<p><em>a</em></p>
</blockquote>
<p>foo <code>bar</code> baz</p>"
`;

exports[`inline 1`] = `
"<p>*a*</p>
<blockquote>
<p>*a*</p>
</blockquote>
<p>foo <code>bar</code> baz</p>"
`;

exports[`inline throws 1`] = `"nope"`;

exports[`regression: actually turn off the tokenizer #412 1`] = `
"<p>hello
world</p>"
`;

exports[`unknown tokenizer 1`] = `
"<p><em>a</em></p>
<blockquote>
<p><em>a</em></p>
</blockquote>
<p>foo <code>bar</code> baz</p>"
`;
